lib/utilities/healthcheckHandler.js

```diff
@@ -45,16 +45,26 @@ export function clientCheck(log, cb) {
 }
 
 function routeHandler(deep, req, res, log, statsClient, cb) {
-    if (isHealthy()) {
-        if (req.method === 'GET' || req.method === 'POST') {
-            if (deep) {
-                return clientCheck(log, cb);
-            }
-            return statsClient.getStats(log, cb);
-        }
+    if (!isHealthy()) {
+        return cb(errors.InternalError, []);
+    }
+    if (req.method !== 'GET' && req.method !== 'POST') {
         return cb(errors.BadRequest, []);
     }
-    return cb(errors.InternalError, []);
+    if (!deep) {
+        return statsClient.getStats(log, cb);
+    }
+    let fail = false;
+    return clientCheck(log, (err, results) => {
+        // makes S3 return 500 error if any backend fails
+        const obj = Object.assign.apply(
+            Object, [{}].concat(results));
```

Comment:
null works
```js
 Object.assign.apply(null, [{}].concat(results));
```
reduce works
```js
results.reduce((obj, item) => Object.assign(obj, item), {})
```
I just used the implementation per transpiling here http://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=latest%2Creact%2Cstage-2&experimental=false&loose=false&spec=false&code=const%20arr%20%3D%20%5B%7B%20a%3A%201%7D%2C%20%7B%20b%3A%202%7D%2C%20%7Bc%3A%203%7D%5D%3B%0AObject.assign(%7B%7D%2C%20...arr)%0A&playground=true